aead/src/aead_key_templates.rs

```diff
@@ -82,6 +82,16 @@ pub fn aes256_ctr_hmac_sha256_key_template() -> KeyTemplate {
     create_aes_ctr_hmac_aead_key_template(32, 16, 32, 32, HashType::Sha256)
 }
 
+/// Return a [`KeyTemplate`] that generates an AES-CTR-HMAC-AEAD key with the following parameters:
+///  - AES key size: 32 bytes
+///  - AES CTR IV size: 16 bytes
+///  - HMAC key size: 32 bytes
```

Comment:
Should HMAC with Sha512 not use a 64 byte key and produce a 64 byte tag?

Reply:
Good spot! (I think the tag length could be shorter but might as well use the full tag)